lib/lightning_web/controllers/collections_controller.ex

```diff
@@ -66,55 +77,177 @@ defmodule LightningWeb.CollectionsController do
     )
   end
 
-  def all(conn, %{"collection" => collection}) do
-    with {:ok, collection} <- Collections.get_collection(collection),
+  #
+  # Controller starts here
+  #
+  def put(conn, %{"name" => col_name, "key" => key, "value" => value}) do
+    with {:ok, collection} <- Collections.get_collection(col_name),
          :ok <- authorize(conn, collection) do
-      conn =
-        conn
-        |> put_resp_content_type("application/json")
-        |> send_chunked(200)
-
-      collection_items = collection_stream(collection)
-
-      # Take a stream of items (they enter the stream in batches of @query_all_limit)
-      # we then chunk them into groups of 20 and send them to the client
-      # since a chunk for each line seems a bit too many
-      # we can play/adjust this number to see what works best
-      wrap_array(collection_items, &Jason.encode!/1)
-      |> Stream.chunk_every(20)
-      |> Enum.reduce_while(conn, fn chunk, conn ->
-        case Plug.Conn.chunk(conn, chunk) do
-          {:ok, conn} ->
-            {:cont, conn}
-
-          {:error, :closed} ->
-            {:halt, conn}
-        end
-      end)
+      case Collections.put(collection, key, value) do
+        :ok ->
+          json(conn, %{key: key, error: nil})
+
+        {:error, :not_foud} ->
+          conn
+          |> put_status(:not_found)
+          |> json(%{key: key, error: "Item Not Found"})
+      end
     end
   end
 
-  defp wrap_array(stream, encode_func) do
-    Stream.concat([
-      ["["],
-      stream
-      |> Stream.map(fn item -> encode_func.(item) end)
-      |> Stream.intersperse(","),
-      ["]"]
-    ])
-  end
-
-  defp collection_stream(collection) do
-    Stream.unfold(nil, fn cursor ->
-      Repo.transaction(fn ->
-        Collections.stream_all(collection, cursor)
-        |> Enum.to_list()
-      end)
-      |> case do
```

Comment:
The starting point for this change is that the records from database to backend are streamed by design. In order to make some good use of it, the database stream would consume it as a stream, without the need to reduce it to a list. 

The repo transaction concern was taken in consideration 4 aspects:
- as any other responsible solution, as the one we have in the original PR, the amount of data fetched is limited per transaction.
- the data is sent via private networking
- the total amount of data on the response is limited on 1000 records. For this amount, it would be acked by worker host unless we get all worker working threads simultaneously hanging.
- There is a limited amount of workers around, on stress tests the max it was setup was 20.

But you have a point on making the capacity planning simpler so I will just remove the streaming from the database. For 1k or 10k records we can keep it in memory like we do now with the reduction to list.